### ISSUE TITLE
`comment-without-dot` reports a comment ending with a colon while its own message says a colon is allowed

### ISSUE THREAD

**Issue by gemshrine** [objectionary/lints#1367]
`comment-without-dot` reports a comment that ends with a colon, and then tells the author that a colon is one of the endings it accepts.

The message, `src/main/resources/org/eolang/lints/comments/comment-without-dot.xsl:34`:

```
The comment for the object doesn't end with a dot (or '?', '!', ':')
```

The predicate, four lines above, knows nothing about a colon:

```xml
<xsl:for-each select="/object/comments/comment[not(ends-with(., '.')) and not(ends-with(., '?')) and not(ends-with(., '!'))]">
```

So this input is reported:

```
# This comment ends with a colon:

[] > foo
  52 > @
```

Verified by adding it as a pack under `comment-without-dot` with `defects: 0`, which fails on master:

```
Tests run: 489, Failures: 1
```

while the two neighbouring packs, `allows-comment-with-question-mark.yaml` and `allows-comment-with-exclamation-mark.yaml`, pass — those two endings really are accepted.

An author who ends a sentence with a colon is told to add one of four endings, one of which is the ending already there.

## Root cause

The message and the predicate were written from different lists. The motive, `src/main/resources/org/eolang/motives/comments/comment-without-dot.md`, says only "Comments must end with a period", so a third source disagrees with both.

## Suggested fix

Decide which list is right and make the other two match. If a colon is a legal ending, the predicate needs it:

```xml
<xsl:for-each select="/object/comments/comment[not(matches(., '[.?!:]$'))]">
```

If it is not, the message should stop offering it:

```
The comment for the object doesn't end with a dot (or '?', '!')
```

A pack for the colon, whichever way it goes, keeps the two from drifting apart again. The motive is worth a line either way, since "must end with a period" already fails to mention `?` and `!`, which the rule has accepted all along.

**Comment by yegor256** [comment]
This was already fixed as a side effect of #1413 (which closed #1384): `comment-without-dot.xsl` now has `not(ends-with(., ':'))` in its predicate, and a pack (`packs/single/comment-without-dot/allows-comment-with-colon.yaml`) already covers the colon case with `defects: 0`. Verified against current `master` — closing as already resolved.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by 0crat** [comment]
@gemshrine Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+3818](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).

**Comment by 0crat** [comment]
@yegor256 Thanks for closing this issue! You've earned +12 points for this: +8 as a basis; +4 for resolving it in less than 12 hours. Your running score is [+2418](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
